### PR TITLE
Publish alpine container

### LINKED ISSUE #288: Tidy up Dockerfiles and remove dependency on drone/ca-certs
I would love to test out the UI rewrite, but I am confused by the seemingly duplicate Dockerfiles. 

E. g. `Dockerfile.agent{.alpine}`  seem quite similar.

The `Dockerfile.agent` apparently still depends on `drone/ca-certs`, which I think should be dropped.

### PR DESCRIPTION
**Pull request by 6543**
* Enable build & publish of alpine docker container
 * Delete outdated BUILD file
 * Move Dockerfiles into subdir "docker"
 * Update & format Dockerfiles
 * Introduce "next" tag

close #288

### CODE REVIEW

**Comment by anbraten** [comment]
@6543 What about #257?

**Comment by 6543** [comment]
@anbraten I'll rebase that one

**Comment by 6543** [comment]
PS: added you as co-author in the squash-merge since took some parts from #257 ...

File: .woodpecker/main.yml [hunk 1]
```diff
@@ -50,84 +50,86 @@ pipeline:
...
     when:
       branch: master
       event: push
 
+  publish-server-alpine:
+    image: plugins/docker
+    repo: woodpeckerci/woodpecker-server
+    dockerfile: docker/Dockerfile.server.alpine
+    secrets: [ docker_username, docker_password ]
+    tag: [latest-alpine]
```
Comment (anbraten) [inline .woodpecker/main.yml:72]: I would change it to `next-alpine` as latest is normally the latest stable release.
```suggestion
    tag: [next-alpine]
```

File: .woodpecker/main.yml [hunk 2]
```diff
@@ -50,84 +50,86 @@ pipeline:
...
   build-cli:
     image: golang:1.16
     commands: make release-cli
 
   publish-server:
     image: plugins/docker
     repo: woodpeckerci/woodpecker-server
+    dockerfile: docker/Dockerfile.server
     secrets: [docker_username, docker_password]
     tag: [latest]
```
Comment (anbraten) [inline .woodpecker/main.yml:62]:
```suggestion
    tag: [next]
```

File: .woodpecker/main.yml [hunk 3]
```diff
@@ -50,84 +50,86 @@ pipeline:
...
-  #   secrets: [ docker_username, docker_password ]
-  #   tag: "${DRONE_TAG}-alpine"
-  #   when:
-  #     event: tag
+  publish-agent-alpine:
+    image: plugins/docker
+    repo: woodpeckerci/woodpecker-agent
+    dockerfile: docker/Dockerfile.agent.alpine
+    secrets: [ docker_username, docker_password ]
+    tag: [latest-alpine]
```
Comment (anbraten) [inline .woodpecker/main.yml:92]:
```suggestion
    tag: [next-alpine]
```

File: .woodpecker/main.yml [hunk 4]
```diff
@@ -50,84 +50,86 @@ pipeline:
...
     tag: ${DRONE_TAG}
     when:
       event: tag
 
+  release-server-alpine:
+    image: plugins/docker
+    repo: woodpeckerci/woodpecker-server
+    dockerfile: docker/Dockerfile.server.alpine
+    secrets: [ docker_username, docker_password ]
+    tag: "${DRONE_TAG}-alpine"
```
Comment (anbraten) [inline .woodpecker/main.yml:111]:
```suggestion
    tag: [latest-alpine, "${DRONE_TAG}-alpine"]
```

File: .woodpecker/main.yml [hunk 5]
```diff
@@ -50,84 +50,86 @@ pipeline:
...
     tag: ${DRONE_TAG}
     when:
       event: tag
 
+  release-agent-alpine:
+    image: plugins/docker
+    repo: woodpeckerci/woodpecker-agent
+    dockerfile: docker/Dockerfile.agent.alpine
+    secrets: [ docker_username, docker_password ]
+    tag: "${DRONE_TAG}-alpine"
```
Comment (anbraten) [inline .woodpecker/main.yml:129]:
```suggestion
    tag: [latest-alpine, "${DRONE_TAG}-alpine"]
```

File: .woodpecker/main.yml [hunk 6]
```diff
@@ -50,84 +50,86 @@ pipeline:
...
+    when:
+      event: tag
+
   release-agent:
     image: plugins/docker
     repo: woodpeckerci/woodpecker-agent
-    dockerfile: Dockerfile.agent
+    dockerfile: docker/Dockerfile.agent
     secrets: [docker_username, docker_password]
     tag: ${DRONE_TAG}
```
Comment (anbraten) [inline .woodpecker/main.yml:120]:
```suggestion
    tag: [latest, ${DRONE_TAG}]
```

**Comment by 6543** [comment]
we can do the latest is latest stable once we did release v0.15.0 else we would have to downgrade current latest witch would break installations unnecessary who did just migrate breaking changes etc ...

**Comment by anbraten** [comment]
Okay I think we could start with next and keep latest until 0.15.0 got released on master pushes as well.

File: .woodpecker/main.yml [hunk 7]
```diff
@@ -50,81 +50,86 @@ pipeline:
...
+    when:
+      branch: master
+      event: push
 
   release-server:
     image: plugins/docker
     repo: woodpeckerci/woodpecker-server
+    dockerfile: docker/Dockerfile.server
     secrets: [docker_username, docker_password]
     tag: ${DRONE_TAG}
```
Comment (6543) [inline .woodpecker/main.yml:105]:
```suggestion
    tag: [latest, "${DRONE_TAG}"]
```

File: .woodpecker/main.yml [hunk 8]
```diff
@@ -50,81 +50,86 @@ pipeline:
...
     image: golang:1.16
     commands: make release-cli
 
   publish-server:
     image: plugins/docker
     repo: woodpeckerci/woodpecker-server
+    dockerfile: docker/Dockerfile.server
     secrets: [docker_username, docker_password]
-    tag: [latest]
+    # TODO: only release as next-alpine after 0.15.0 got released
```
Comment (6543) [inline .woodpecker/main.yml:62]:
```suggestion
    # TODO: only release as next after 0.15.0 got released
```

File: .woodpecker/main.yml [hunk 9]
```diff
@@ -50,81 +50,86 @@ pipeline:
...
       branch: master
       event: push
 
   publish-agent:
     image: plugins/docker
     repo: woodpeckerci/woodpecker-agent
-    dockerfile: Dockerfile.agent
+    dockerfile: docker/Dockerfile.agent
     secrets: [docker_username, docker_password]
     tag: [latest]
```
Comment (6543) [inline .woodpecker/main.yml:84]:
```suggestion
    # TODO: only release as next after 0.15.0 got released
    tag: [next, latest]
```

**Review by 6543 — COMMENTED** [review]
lgtm

**Comment by anbraten** [comment]
Okay almost. Could you add next tag to `/docs/downloads` with a small description on latest vs next?

**Comment by anbraten** [comment]
Ahh and we need to fix `.dockerignore`. Atm it pushes all files to the build context. I fixed that in #257 by only adding the dockerfile and binary to a folder used as context. [see here](https://github.com/woodpecker-ci/woodpecker/pull/257/files#diff-108c5a32c51adb7b88859120d9a3f2e6d877515a71c37db3155f4b6bfbd99d5bR19-R21)

**Comment by 6543** [comment]
> Ahh and we need to fix `.dockerignore`. Atm it pushes all files to the build context. I fixed that in #257 by only adding the dockerfile and binary to a folder used as context. [see here](https://github.com/woodpecker-ci/woodpecker/pull/257/files#diff-108c5a32c51adb7b88859120d9a3f2e6d877515a71c37db3155f4b6bfbd99d5bR19-R21)

I think we dont need that, it's not within the final image ...

**Review by anbraten — APPROVED** [review]

**Pull request merged**